src/Alfred/Cache.hs

```diff
@@ -1,11 +1,34 @@
 module Alfred.Cache
     ( writeItems
+    , rawWriteItems
     ) where
 
 import qualified Data.Csv as Csv
 import qualified Data.ByteString.Lazy as BS
 
 import Alfred.Item
+import Alfred.Search
+import qualified Data.HashMap.Strict as HM
 
 writeItems :: String -> [Item] -> IO ()
-writeItems cacheFile = BS.writeFile cacheFile . Csv.encodeDefaultOrderedByName
+writeItems cacheFile items = do
+  cachedItems <- getItems cacheFile
+  rawWriteItems cacheFile $ itemsWithCachedVisits items cachedItems
+
+rawWriteItems :: String -> [Item] -> IO ()
```

Comment:
The reason why I'd done this, also, was to maintain the external public interface. If there's a better way to do this, or if this should be a separate API and we expect each library using this to update, that works too.